src/GraphQL/Utilities/EmptyEnumerator.cs

```diff
@@ -0,0 +1,22 @@
+using System.Collections;
+using System.Collections.Generic;
+
+namespace GraphQL.Utilities
+{
+    internal sealed class EmptyEnumerator<T> : IEnumerator<T>
```

Comment:
Doesn't Enumerable.Empty<> work the same?  See https://github.com/dotnet/runtime/blob/master/src/libraries/System.Linq/src/System/Linq/Enumerable.SpeedOpt.cs

Reply:
Finally found the EmptyPartition class: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Linq/src/System/Linq/Partition.SpeedOpt.cs

Reply:
ok